src/Symfony/Component/Translation/Extractor/PhpAstExtractor.php

```diff
@@ -39,7 +40,11 @@ public function __construct(
             throw new \LogicException(sprintf('You cannot use "%s" as the "nikic/php-parser" package is not installed. Try running "composer require nikic/php-parser".', static::class));
         }
 
-        $this->parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+        if (method_exists(ParserFactory::class, 'createForVersion')) {
+            $this->parser = (new ParserFactory())->createForVersion(PhpVersion::fromString('8.2'));
```

Comment:
I'm wondering how we will maintain this version number.
Wild idea: add // @requires PHP 8.2 so that grepping for test cases can yield this when preparing 8.0?

Reply:
Not sure either, but one possible solution would be to use `createForHostVersion()` instead which will simply use the PHP version the process is executed with. Probably that's also how people most often use this extractor anyway.

Reply:
replaced with `createForHostVersion()` as this is also basically what @stof asks for in https://github.com/symfony/symfony/pull/53453#discussion_r1444332797